web/wai-server/Main.hs

```diff
@@ -66,15 +66,15 @@ socketsApp ref = websocketsOr defaultConnectionOptions wsApp defaultApp where
     sendTextData conn (B.pack $ show i)
     putMVar convar (name, conn)
     forkPingThread conn 30
-    forever $ receiveMove name conn -- listen for moves, forever.
+    forever $ receiveMove (read $ show i :: Int) conn -- listen for moves, forever.
```

Comment:
So, I think we should just change the signature of the connect function to take an `IORef Int` and an `Int` instead of `IORef a` and an `a`, that way we won't have to deal with showing and reading. I had originally left the function with it's most generic possible signature, but now that signature imposes non type safe operations. One the signature is fixed, we can just pass `i` directly to `receiveMode`.

What do you think?